<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sysbiolab Team" />

<meta name="date" content="2025-10-17" />

<title>Creating signal aggregation rules</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
<link href="site_libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script>
<script src="site_libs/bs3compat-0.9.0/transition.js"></script>
<script src="site_libs/bs3compat-0.9.0/tabs.js"></script>
<script src="site_libs/bs3compat-0.9.0/bs3compat.js"></script>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PathwaySpace</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="install.html">Install</a>
</li>
<li>
  <a href="get-started.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Introductory Vignettes</li>
    <li>
      <a href="projection-methods.html">PathwaySpace projection methods</a>
    </li>
    <li>
      <a href="modeling-signal-decay.html">Modeling signal decay functions</a>
    </li>
    <li>
      <a href="signal-aggregation-rules.html">Creating signal aggregation rules</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Visualizing features</li>
    <li>
      <a href="sparse-feature-sets.html">Sparse feature sets</a>
    </li>
    <li>
      <a href="spatial-transcriptomics.html">Spatial transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/sysbiolab/PathwaySpace">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Creating signal aggregation rules</h1>
<h4 class="author">Sysbiolab Team</h4>
<h4 class="date">2025-10-17</h4>

</div>


<p><strong>Package</strong>: PathwaySpace 1.0.3.1 <br/></p>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>When aggregating multiple signals over a space, such as
<em>PathwaySpace</em> projections, it’s often not enough to simply add
or average their values, given that each signal carries uncertainty or
partial information.</p>
<p>For practical implementation with <em>PathwaySpace</em> projections,
any unary aggregation function of the form
<strong><code>function(x) { ... }</code></strong> can be plugged into
the pipeline, which has essentially two steps: (i) map raw signal
intensities to a decay function, such as linear, exponential, or
sigmoidal, depending on the data (see <a
href="modeling-signal-decay.html"><em>modeling signal decay</em></a>);
and (ii) use an appropriate operation to produce a combined projection.
The decay function models the signal propagation, each vertex emitting a
projection that decreases with distance, reflecting both the local
strength of the signal and its spatial influence. Because a graph may
have multiple vertices, the resulting projections can be aggregated in
different ways (see <a href="index.html"><strong>Figure 1</strong></a>
for a conceptual overview of signal processing in
<em>PathwaySpace</em>).</p>
<p>In what follows, this tutorial demonstrates how to create different
signal aggregation functions for <em>PathwaySpace</em> projections. A
toy graph with five small islands is used to illustrate aggregation
rules that emphasize distinct patterns in the data. The final section
suggests a list of <em>fuzzy logic</em> operators that can be use to
aggregate signals in <em>PathwaySpace</em>, including the trade-offs of
each approach.</p>
</div>
<div id="required-packages" class="section level1">
<h1>Required packages</h1>
<pre class="r"><code># Check required packages for this vignette
if (!require(&quot;remotes&quot;, quietly = TRUE)){
  install.packages(&quot;remotes&quot;)
}
if (!require(&quot;RGraphSpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (!require(&quot;PathwaySpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}
if (!require(&quot;SpotSpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/SpotSpace&quot;)
}</code></pre>
<pre class="r"><code># Check versions
if (packageVersion(&quot;RGraphSpace&quot;) &lt; &quot;1.1.0&quot;){
  message(&quot;Need to update &#39;RGraphSpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (packageVersion(&quot;PathwaySpace&quot;) &lt; &quot;1.0.3.1&quot;){
  message(&quot;Need to update &#39;PathwaySpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}
if (packageVersion(&quot;SpotSpace&quot;) &lt; &quot;0.0.2&quot;){
  message(&quot;Need to update &#39;SpotSpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/SpotSpace&quot;)
}</code></pre>
<pre class="r"><code># Load packages
library(igraph)
library(ggplot2)
library(RGraphSpace)
library(PathwaySpace)
library(SpotSpace)
library(patchwork)</code></pre>
</div>
<div id="setting-basic-input-data" class="section level1">
<h1>Setting basic input data</h1>
<pre class="r"><code># Create a simple graph with five islands
g &lt;- rep(igraph::make_star(7, &quot;undirected&quot;), 5)
V(g)$name &lt;- paste0(&quot;n&quot;,1:vcount(g))
gs &lt;- GraphSpace(g, layout = layout_nicely(g))
plotGraphSpace(gs, add.labels = TRUE)</code></pre>
<p><img src="signal-aggregation-rules_files/figure-html/Modeling%20signal%20decay%20-%201-1.png" width="672" /></p>
<pre class="r"><code># Build a PathwaySpace object
ps &lt;- buildPathwaySpace(gs)</code></pre>
</div>
<div id="assigning-a-decay-model" class="section level1">
<h1>Assigning a decay model</h1>
<p>Next, we will assign a linear decay model for all vertices (for
additional details, see the <a
href="modeling-signal-decay.html"><em>modeling signal decay</em></a>
tutorial).</p>
<pre class="r"><code># Get distance to the nearest nodes
near_df &lt;- getNearestNode(ps)
pdist &lt;- mean(near_df$dist)
# The &#39;pdist&#39; is the average center-to-center distance between nodes
pdist</code></pre>
<pre><code>## [1] 0.1017494</code></pre>
<pre class="r"><code># Setting a linear decay model for all vertices
vertexDecay(ps) &lt;- linearDecay(pdist = pdist)</code></pre>
<p>As a default behavior, the <code>buildPathwaySpace()</code>
constructor initializes each vertex’s signal with <code>0</code> (for
more details, see the <a
href="projection-methods.html#Signal_types"><em>signal types</em></a>
section). We will update this signal, setting peak values at the central
vertices or hubs.</p>
<pre class="r"><code># Find hubs 
hubs &lt;- degree(ps@graph)
hubs &lt;- hubs[hubs&gt;1]
# Add signal
vertexSignal(ps) &lt;- 1
vertexSignal(ps)[names(hubs)] &lt;- 2
# Vertex count
gs_vcount(ps)</code></pre>
<pre><code>## [1] 35</code></pre>
</div>
<div id="creating-aggregation-rules" class="section level1">
<h1>Creating aggregation rules</h1>
<div id="means-max-and-min" class="section level2">
<h2>Means, max, and min</h2>
<p>Now we project the signals using the
<code>circularProjection()</code> function, in which the
<code>aggregate.fun</code> argument defines the aggregation rule. This
argument accepts any user-defined unary aggregation function that
aggregates a vector of numeric values into a single scalar output.</p>
<p>In the examples below, we compare two approaches: a standard
<em>arithmetic mean</em>, where all values contribute equally, and a
<em>self-weighted (contraharmonic) mean</em>, where each value is
weighted by its own magnitude.</p>
<pre class="r"><code># Signal aggregation by &#39;mean&#39;
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = mean)
p1 &lt;- plotPathwaySpace(ps, theme = &quot;th3&quot;, title = &quot;mean()&quot;)

# Signal aggregation by a &#39;self-weighted mean&#39;
# (generalized contraharmonic to -/+ values)
wmean &lt;- function(x){ weighted.mean(x, abs(x)) }
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = wmean)
p2 &lt;- plotPathwaySpace(ps, theme = &quot;th3&quot;, title = &quot;wmean()&quot;)

p1 + p2</code></pre>
<p><img src="signal-aggregation-rules_files/figure-html/Modeling%20signal%20decay%20-%206-1.png" width="100%" /></p>
<p>These examples show how changes in weighting (from equal weights to
self-weights) affects the combined projection: The standard
<em>arithmetic mean</em> may be used when we want to emphasize the
influence of local groups, while the <em>self-weighted mean</em> will
shift the mean toward the more dominant signals.</p>
<p>Next, we assign new aggregation functions to explore different rules.
The <code>max</code> function also emphasizes the dominant signals while
the <code>min</code> function will highlight the regions where
projections intersect between neighbor vertices.</p>
<pre class="r"><code># Signal aggregation by &#39;max&#39;
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = max)
p1 &lt;- plotPathwaySpace(ps, theme = &quot;th3&quot;, title = &quot;max()&quot;)

# Signal aggregation by &#39;min&#39;
ps &lt;- circularProjection(ps, k = 2, aggregate.fun = min)
p2 &lt;- plotPathwaySpace(ps, theme = &quot;th3&quot;, title = &quot;min()&quot;)

p1 + p2</code></pre>
<p><img src="signal-aggregation-rules_files/figure-html/Modeling%20signal%20decay%20-%207-1.png" width="100%" /></p>
<p>Note that the <code>k</code> parameter affects the aggregations, as
it determines how many projected signals are considered at each point in
space. In general, we recommend using <code>k = gs_vcount(ps)</code> to
capture contributions from all vertices, except in special cases, such
as the example above using the <code>min</code> function, where the
aggregation aims to highlight minimal values between two neighbor
vertices.</p>
</div>
<div id="fuzzy-unions-and-intersections" class="section level2">
<h2>Fuzzy unions and intersections</h2>
<p>This section lists <em>fuzzy logic</em> functions that can be use to
aggregate signals in <em>PathwaySpace</em>, mostly available from the
<em>lfl</em> package <span class="citation">(Burda and Štěpnička
2021)</span>. A fuzzy intersection measures how strongly all signals
agree at a given point, typically using a <em>t-norm</em> like minimum
or product. A fuzzy union, in contrast, measures how strongly any of the
signals are at a given point, using a <em>t-conorm</em> like maximum. By
combining fuzzy unions or intersections with decay functions, we can
produce unified projections that integrate multiple signal sources into
a single, interpretable representation.</p>
<pre class="r"><code>if (!require(&quot;lfl&quot;, quietly = TRUE)){
  install.packages(&quot;lfl&quot;)
}
# Load Fuzzy Logic functions
library(lfl)</code></pre>
<p></br></p>
<div id="t-norms-triangular-norms" class="section level3">
<h3>T-norms (Triangular Norms)</h3>
<p>These functions represent fuzzy “AND” operations:</p>
<pre class="r"><code># Goedel t-norm: use when aggregation should be strictly limited by the weakest
# signal (logical AND, conservative intersection);
# Input range in [0, 1]
ps &lt;- circularProjection(ps, k = 2, aggregate.fun = goedel.tnorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Goguen t-norm: use when you want smooth multiplicative attenuation of 
# overlapping signals (probabilistic AND);
# Input range in [0, 1]
ps &lt;- circularProjection(ps, k = 2, aggregate.fun = goguen.tnorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Hamacher product t-norm: use when you need smoothness in signal intersections;
# Input range in [-Inf, Inf];
fuzzy.hamacher.tnorm &lt;- function(x) {
  Reduce(function(a, b) (a * b) / (a + b - a * b), x)
}
ps &lt;- circularProjection(ps, k = 2, aggregate.fun = fuzzy.hamacher.tnorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Einstein product t-norm: use when combining signals should preserve low values
# without collapsing to zero;
# Input range in [-Inf, Inf]
fuzzy.einstein.tnorm &lt;- function(x) {
  Reduce(function(a, b) (a * b) / (1 + (1 - a) * (1 - b)), x)
}
ps &lt;- circularProjection(ps, k = 2, aggregate.fun = fuzzy.einstein.tnorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)</code></pre>
<p><img src="figs_spot_space/fig_b1.png" width="100%" /></p>
<p></br></p>
</div>
<div id="t-conorms-triangular-conorms" class="section level3">
<h3>T-conorms (Triangular Conorms)</h3>
<p>These functions represent fuzzy “OR” operations:</p>
<pre class="r"><code># Goedel t-conorm: use when you want an extremely permissive OR, where the 
# strongest signal fully dominates the aggregation;
# Input range in [0, 1]
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = goedel.tconorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Goguen t-conorm: use when overlapping signals should reinforce each other in 
# a probabilistic way (soft OR);
# Input range in  [0, 1]
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = goguen.tconorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Lukasiewicz t-conorm: use when you want signal reinforcement, saturating at 1 
# once contributions accumulate;
# Input range in [0, 1]
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = lukas.tconorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Hamacher sum t-conorm: use when you prefer a moderate, nonlinear signal reinforcement 
# that avoids rapid saturation;
# Input range in [0, 1]
fuzzy.hamacher.tconorm &lt;- function(x) {
  stopifnot(is.numeric(x), all(x &gt;= 0), all(x &lt;= 1))
  Reduce(function(a, b) (a + b - 2 * a * b) / (1 - a * b), x)
}
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = fuzzy.hamacher.tconorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)

# Einstein sum t-conorm: use when signal reinforcement should be smooth and symmetric,
# limiting extreme increases;
# Input range in [-Inf, Inf]
fuzzy.einstein.tconorm &lt;- function(x) {
  Reduce(function(a, b) (a + b) / (1 + a * b), x)
}
ps &lt;- circularProjection(ps, k = gs_vcount(ps), aggregate.fun = fuzzy.einstein.tconorm)
plotPathwaySpace(ps, theme = &quot;th3&quot;)</code></pre>
<p><img src="figs_spot_space/fig_b2.png" width="100%" /></p>
<p>Note that most of the fuzzy functions listed above, including
<em>t-norms</em> (e.g., Goedel, Goguen, Hamacher) and <em>t-conorms</em>
(e.g., Goedel, Goguen, Lukasiewicz), operate within the unit interval,
accepting input values in <code>[0, 1]</code>. <em>PathwaySpace</em>
also operate on this range of values for positive signals by default
(with <code>rescale = TRUE</code>; see <code>circularProjection()</code>
function); however, negative signals are rescaled to
<code>[-1, 0]</code>, and therefore are not directly compatible with
these fuzzy operators (for more details, see the <a
href="projection-methods.html#Signal_types"><em>signal types</em></a>
section).</p>
<p>Also, when evaluating input signals in the range
<code>(-Inf, Inf)</code> (<em>e.g.</em>, differential expression
values), we recommend using aggregation functions that treat negative
and positive values symmetrically. In such cases, the standard
<em>arithmetic mean</em> and <em>self-weighted mean</em> provide
compatible sign-symmetric aggregation strategies.</p>
</div>
</div>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<pre><code>## R version 4.5.1 (2025-06-13)
## Platform: x86_64-pc-linux-gnu
## Running under: Ubuntu 24.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: America/Sao_Paulo
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] patchwork_1.3.2      igraph_2.2.0         SpotSpace_0.0.2     
## [4] PathwaySpace_1.0.3.1 RGraphSpace_1.1.0    ggplot2_4.0.0.9000  
## [7] remotes_2.5.0        bs4cards_0.1.1      
## 
## loaded via a namespace (and not attached):
##   [1] deldir_2.0-4           pbapply_1.7-4          gridExtra_2.3         
##   [4] rlang_1.1.6            magrittr_2.0.4         RcppAnnoy_0.0.22      
##   [7] spatstat.geom_3.6-0    matrixStats_1.5.0      ggridges_0.5.7        
##  [10] compiler_4.5.1         png_0.1-8              vctrs_0.6.5           
##  [13] reshape2_1.4.4         stringr_1.5.2          pkgconfig_2.0.3       
##  [16] fastmap_1.2.0          labeling_0.4.3         fontawesome_0.5.3     
##  [19] promises_1.3.3         rmarkdown_2.30         purrr_1.1.0           
##  [22] xfun_0.53              cachem_1.1.0           jsonlite_2.0.0        
##  [25] goftest_1.2-3          later_1.4.4            spatstat.utils_3.2-0  
##  [28] irlba_2.3.5.1          parallel_4.5.1         cluster_2.1.8.1       
##  [31] R6_2.6.1               ica_1.0-3              spatstat.data_3.1-8   
##  [34] stringi_1.8.7          bslib_0.9.0            RColorBrewer_1.1-3    
##  [37] reticulate_1.43.0      spatstat.univar_3.1-4  parallelly_1.45.1     
##  [40] lmtest_0.9-40          jquerylib_0.1.4        scattermore_1.2       
##  [43] Rcpp_1.1.0             knitr_1.50             tensor_1.5.1          
##  [46] future.apply_1.20.0    zoo_1.8-14             sctransform_0.4.2     
##  [49] httpuv_1.6.16          Matrix_1.7-4           splines_4.5.1         
##  [52] tidyselect_1.2.1       abind_1.4-8            rstudioapi_0.17.1     
##  [55] yaml_2.3.10            spatstat.random_3.4-2  spatstat.explore_3.5-3
##  [58] codetools_0.2-20       miniUI_0.1.2           listenv_0.9.1         
##  [61] lattice_0.22-5         tibble_3.3.0           plyr_1.8.9            
##  [64] shiny_1.11.1           withr_3.0.2            S7_0.2.0              
##  [67] ROCR_1.0-11            evaluate_1.0.5         Rtsne_0.17            
##  [70] future_1.67.0          fastDummies_1.7.5      survival_3.8-3        
##  [73] polyclip_1.10-7        fitdistrplus_1.2-4     pillar_1.11.1         
##  [76] Seurat_5.3.1.9999      KernSmooth_2.23-26     plotly_4.11.0         
##  [79] generics_0.1.4         RcppHNSW_0.6.0         sp_2.2-0              
##  [82] scales_1.4.0           globals_0.18.0         xtable_1.8-4          
##  [85] glue_1.8.0             lazyeval_0.2.2         tools_4.5.1           
##  [88] data.table_1.17.8      RSpectra_0.16-2        RANN_2.6.2            
##  [91] fs_1.6.6               dotCall64_1.2          cowplot_1.2.0         
##  [94] grid_4.5.1             tidyr_1.3.1            nlme_3.1-168          
##  [97] cli_3.6.5              spatstat.sparse_3.1-0  spam_2.11-1           
## [100] viridisLite_0.4.2      dplyr_1.1.4            uwot_0.2.3            
## [103] gtable_0.3.6           sass_0.4.10            digest_0.6.37         
## [106] progressr_0.17.0       ggrepel_0.9.6          htmlwidgets_1.6.4     
## [109] SeuratObject_5.2.0     farver_2.1.2           htmltools_0.5.8.1     
## [112] lifecycle_1.0.4        httr_1.4.7             mime_0.13             
## [115] MASS_7.3-65</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-Burda2021" class="csl-entry">
Burda, Michal, and Martin Štěpnička. 2021. <span>“Lfl: An <span>R</span>
Package for Linguistic Fuzzy Logic.”</span> <em>Fuzzy Sets and
Systems</em>. <a
href="https://doi.org/10.1016/j.fss.2021.07.007">https://doi.org/10.1016/j.fss.2021.07.007</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
