<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sysbiolab Team" />

<meta name="date" content="2026-02-27" />

<title>Visualizing sparse feature sets on large graphs</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
<link href="site_libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script>
<script src="site_libs/bs3compat-0.9.0/transition.js"></script>
<script src="site_libs/bs3compat-0.9.0/tabs.js"></script>
<script src="site_libs/bs3compat-0.9.0/bs3compat.js"></script>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PathwaySpace</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="install.html">Install</a>
</li>
<li>
  <a href="get-started.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Introductory Vignettes</li>
    <li>
      <a href="projection-methods.html">PathwaySpace projection methods</a>
    </li>
    <li>
      <a href="modeling-signal-decay.html">Modeling signal decay functions</a>
    </li>
    <li>
      <a href="signal-aggregation-rules.html">Creating signal aggregation rules</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Visualizing features</li>
    <li>
      <a href="sparse-feature-sets.html">Sparse feature sets</a>
    </li>
    <li>
      <a href="spatial-transcriptomics.html">Spatial transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/sysbiolab/PathwaySpace">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Visualizing sparse feature sets on large
graphs</h1>
<h4 class="author">Sysbiolab Team</h4>
<h4 class="date">2026-02-27</h4>

</div>


<p><strong>Package</strong>: PathwaySpace 1.1.1 <br/></p>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This tutorial creates a large <em>PathwaySpace</em> object with
<code>n = 12990</code> vertices, upon which we will project binary
signals representing feature sets from a relatively small number of
vertices. The goal is to enhance clarity and make it less likely for
viewers to miss important details of large graphs when only a limited
number of features carry relevant information. The projections will
emphasize clusters of vertices forming <em>summits</em>, and we will add
silhouettes as decorative elements to outline the overall graph
structure. The examples in this section are adapted from <span
class="citation">Ellrott et al. (2025)</span> and <span
class="citation">Tercan et al. (2025)</span>.</p>
<p>We will start by loading an <em>igraph</em> object containing gene
interaction data available from the <em>Pathway Commons</em> database
(version 12) <span class="citation">(Rodchenkov et al. 2019)</span>.</p>
</div>
<div id="required-packages" class="section level1">
<h1>Required packages</h1>
<pre class="r"><code># Check required packages for this vignette
if (!require(&quot;remotes&quot;, quietly = TRUE)){
  install.packages(&quot;remotes&quot;)
}
if (!require(&quot;RGraphSpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (!require(&quot;PathwaySpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}</code></pre>
<pre class="r"><code># Check versions
if (packageVersion(&quot;RGraphSpace&quot;) &lt; &quot;1.1.1&quot;){
  message(&quot;Need to update &#39;RGraphSpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (packageVersion(&quot;PathwaySpace&quot;) &lt; &quot;1.1.1&quot;){
  message(&quot;Need to update &#39;PathwaySpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}</code></pre>
<pre class="r"><code># Load packages
library(PathwaySpace)
library(RGraphSpace)
library(igraph)
library(ggplot2)</code></pre>
</div>
<div id="setting-input-data" class="section level1">
<h1>Setting input data</h1>
<pre class="r"><code># Load a large igraph object
data(&quot;PCv12_pruned_igraph&quot;, package = &quot;PathwaySpace&quot;)

# Check number of vertices
length(PCv12_pruned_igraph)
# [1] 12990

# Check vertex names
head(V(PCv12_pruned_igraph)$name)
# [1] &quot;A1BG&quot; &quot;AKT1&quot; &quot;CRISP3&quot; &quot;GRB2&quot; &quot;PIK3CA&quot; &quot;PIK3R1&quot;

# Get top-connected nodes for visualization
top10hubs &lt;- igraph::degree(PCv12_pruned_igraph)
top10hubs &lt;- names(sort(top10hubs, decreasing = TRUE)[1:10])
head(top10hubs)
# [1] &quot;GNB1&quot; &quot;TRIM28&quot; &quot;RPS27A&quot; &quot;CTNNB1&quot; &quot;TP53&quot; &quot;ACTB&quot;</code></pre>
<pre class="r"><code>## Check graph validity
g_space_PCv12 &lt;- GraphSpace(PCv12_pruned_igraph, mar = 0.1)</code></pre>
<pre class="r"><code>## Visualize the graph layout labeled with &#39;top10hubs&#39; nodes
plotGraphSpace(g_space_PCv12, node.labels = top10hubs, label.color = &quot;blue&quot;, theme = &quot;th3&quot;)</code></pre>
<p><img src="figs_feature_sets/fig1.png" width="90%" /></p>
<p>We now load gene sets from the <em>MSigDB</em> collection <span
class="citation">(Liberzon et al. 2015)</span>, which are subsequently
used to project a binary signal onto the <em>PathwaySpace</em>
image.</p>
<pre class="r"><code># Load a list with Hallmark gene sets
data(&quot;Hallmarks_v2023_1_Hs_symbols&quot;, package = &quot;PathwaySpace&quot;)

# There are 50 gene sets in &quot;hallmarks&quot;
length(hallmarks)
# [1] 50

# We will use the &#39;HALLMARK_P53_PATHWAY&#39; (n=200 genes) for demonstration
length(hallmarks$HALLMARK_P53_PATHWAY)
# [1] 200</code></pre>
</div>
<div id="running-pathwayspace" class="section level1">
<h1>Running <em>PathwaySpace</em></h1>
<p>We now follow the <em>PathwaySpace</em> pipeline as explained in the
<a href="projection-methods.html">introductory vignette</a>, that is,
using the <code>buildPathwaySpace()</code> constructor to initialize a
new <em>PathwaySpace</em> object with the <em>Pathway Commons</em>
interactions.</p>
<pre class="r"><code># Run the PathwaySpace constructor
p_space_PCv12 &lt;- buildPathwaySpace(gs=g_space_PCv12, nrc=500)
# Note: &#39;nrc&#39; sets the number of rows and columns of the
# image space, which will affect the image resolution (in pixels)</code></pre>
<p>…and mark the <em>HALLMARK_P53_PATHWAY</em> genes in the
<em>PathwaySpace</em> object.</p>
<pre class="r"><code># Intersect Hallmark genes with the PathwaySpace
hallmarks &lt;- lapply(hallmarks, intersect, y = names(p_space_PCv12) )

# After intersection, the &#39;HALLMARK_P53_PATHWAY&#39; dropped to n=173 genes
length(hallmarks$HALLMARK_P53_PATHWAY)
# [1] 173

# Set a binary signal (1s) to &#39;HALLMARK_P53_PATHWAY&#39; genes
vertexSignal(p_space_PCv12) &lt;- 0
vertexSignal(p_space_PCv12)[ hallmarks$HALLMARK_P53_PATHWAY ] &lt;- 1</code></pre>
<p>…and run the <code>circularProjection()</code> function.</p>
<pre class="r"><code># Run signal projection
p_space_PCv12 &lt;- circularProjection(p_space_PCv12)</code></pre>
<p>Next, we decorate the <em>PathwaySpace</em> image with graph
silhouettes and plot the results.</p>
<pre class="r"><code># Add silhouettes
p_space_PCv12 &lt;- silhouetteMapping(p_space_PCv12)
# Plot the results
plotPathwaySpace(p_space_PCv12, title=&quot;HALLMARK_P53_PATHWAY&quot;, marks = top10hubs, mark.size = 2, theme = &quot;th3&quot;)</code></pre>
<p><img src="figs_feature_sets/fig2.png" width="100%" /></p>
</div>
<div id="mapping-summits" class="section level1">
<h1>Mapping summits</h1>
<p>The summits represent regions within the graph that exhibit signal
values that are notably higher than the baseline level. These regions
may be of interest for downstream analyses. One potential downstream
analysis is to determine which vertices projected the original input
signal. This could provide insights into communities within these summit
regions. One may also wish to explore other vertices within the summits,
by querying associations with the original input gene set. In order to
extract vertices within summits, next we use the
<code>summitMapping()</code> function, which also decorates summits with
contour lines.</p>
<pre class="r"><code># Mapping summits
p_space_PCv12 &lt;- summitMapping(p_space_PCv12, minsize = 50)
plotPathwaySpace(p_space_PCv12, title=&quot;Summit regions&quot;, theme = &quot;th3&quot;)</code></pre>
<p><img src="figs_feature_sets/fig3.png" width="100%" /></p>
<pre class="r"><code># Extracting summits from a PathwaySpace
summits &lt;- getPathwaySpace(p_space_PCv12, &quot;summits&quot;)
class(summits)
# [1] &quot;list&quot;</code></pre>
</div>
<div id="citation" class="section level1">
<h1>Citation</h1>
<p>If you use <em>PathwaySpace</em>, please cite:</p>
<ul>
<li><p>Tercan &amp; Apolonio et al. Protocol for assessing distances in
pathway space for classifier feature sets from machine learning methods.
<em>STAR Protocols</em> 6(2):103681, 2025. <a
href="https://doi.org/10.1016/j.xpro.2025.103681"
class="uri">https://doi.org/10.1016/j.xpro.2025.103681</a></p></li>
<li><p>Ellrott et al. Classification of non-TCGA cancer samples to TCGA
molecular subtypes using compact feature sets. <em>Cancer Cell</em>
43(2):195-212.e11, 2025. <a
href="https://doi.org/10.1016/j.ccell.2024.12.002"
class="uri">https://doi.org/10.1016/j.ccell.2024.12.002</a></p></li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<pre><code>## R version 4.5.2 (2025-10-31)
## Platform: x86_64-pc-linux-gnu
## Running under: Ubuntu 24.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: America/Sao_Paulo
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] patchwork_1.3.2    igraph_2.2.0       SpotSpace_0.0.5    PathwaySpace_1.1.1
## [5] RGraphSpace_1.1.1  ggplot2_4.0.0.9000 remotes_2.5.0      bs4cards_0.1.1    
## 
## loaded via a namespace (and not attached):
##   [1] RColorBrewer_1.1-3     rstudioapi_0.17.1      jsonlite_2.0.0        
##   [4] shape_1.4.6.1          magrittr_2.0.4         spatstat.utils_3.2-0  
##   [7] ggbeeswarm_0.7.2       farver_2.1.2           rmarkdown_2.30        
##  [10] GlobalOptions_0.1.2    fs_1.6.6               vctrs_0.6.5           
##  [13] ROCR_1.0-11            spatstat.explore_3.5-3 htmltools_0.5.8.1     
##  [16] sass_0.4.10            sctransform_0.4.2      parallelly_1.45.1     
##  [19] KernSmooth_2.23-26     bslib_0.9.0            htmlwidgets_1.6.4     
##  [22] ica_1.0-3              fontawesome_0.5.3      plyr_1.8.9            
##  [25] plotly_4.11.0          zoo_1.8-14             cachem_1.1.0          
##  [28] mime_0.13              lifecycle_1.0.4        pkgconfig_2.0.3       
##  [31] Matrix_1.7-4           R6_2.6.1               fastmap_1.2.0         
##  [34] fitdistrplus_1.2-4     future_1.67.0          shiny_1.11.1          
##  [37] digest_0.6.37          colorspace_2.1-1       Seurat_5.4.0.9009     
##  [40] tensor_1.5.1           RSpectra_0.16-2        irlba_2.3.5.1         
##  [43] progressr_0.17.0       spatstat.sparse_3.1-0  httr_1.4.7            
##  [46] polyclip_1.10-7        abind_1.4-8            compiler_4.5.2        
##  [49] withr_3.0.2            S7_0.2.0               fastDummies_1.7.5     
##  [52] MASS_7.3-65            tools_4.5.2            vipor_0.4.7           
##  [55] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.16         
##  [58] future.apply_1.20.0    goftest_1.2-3          glue_1.8.0            
##  [61] nlme_3.1-168           promises_1.3.3         grid_4.5.2            
##  [64] Rtsne_0.17             cluster_2.1.8.2        reshape2_1.4.4        
##  [67] generics_0.1.4         gtable_0.3.6           spatstat.data_3.1-8   
##  [70] tidyr_1.3.1            data.table_1.17.8      sp_2.2-0              
##  [73] spatstat.geom_3.6-0    RcppAnnoy_0.0.22       ggrepel_0.9.6         
##  [76] RANN_2.6.2             pillar_1.11.1          stringr_1.5.2         
##  [79] spam_2.11-1            RcppHNSW_0.6.0         later_1.4.4           
##  [82] circlize_0.4.16        splines_4.5.2          dplyr_1.1.4           
##  [85] lattice_0.22-9         survival_3.8-6         deldir_2.0-4          
##  [88] tidyselect_1.2.1       miniUI_0.1.2           pbapply_1.7-4         
##  [91] knitr_1.50             gridExtra_2.3          scattermore_1.2       
##  [94] xfun_0.54              matrixStats_1.5.0      stringi_1.8.7         
##  [97] lazyeval_0.2.2         yaml_2.3.10            evaluate_1.0.5        
## [100] codetools_0.2-20       tibble_3.3.0           cli_3.6.5             
## [103] uwot_0.2.3             xtable_1.8-4           reticulate_1.44.1     
## [106] jquerylib_0.1.4        dichromat_2.0-0.1      Rcpp_1.1.0            
## [109] globals_0.18.0         spatstat.random_3.4-2  png_0.1-8             
## [112] ggrastr_1.0.2          spatstat.univar_3.1-4  parallel_4.5.2        
## [115] dotCall64_1.2          listenv_0.9.1          viridisLite_0.4.2     
## [118] scales_1.4.0           ggridges_0.5.7         SeuratObject_5.3.0    
## [121] purrr_1.1.0            rlang_1.1.6            cowplot_1.2.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-Ellrott2025" class="csl-entry">
Ellrott, Kyle, Christopher K Wong, Christina Yau, Mauro A A Castro, et
al. 2025. <span>“Classification of Non-TCGA Cancer Samples to TCGA
Molecular Subtypes Using Compact Feature Sets.”</span> <em>Cancer
Cell</em> 43 (2): 195–212. <a
href="https://doi.org/10.1016/j.ccell.2024.12.002">https://doi.org/10.1016/j.ccell.2024.12.002</a>.
</div>
<div id="ref-Liberzon2015" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdottir, Mahmoud Ghandi,
Jill Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures
Database (MSigDB) Hallmark Gene Set Collection.”</span> <em>Cell
Systems</em> 1 (5): 417–25. <a
href="https://doi.org/10.1016/j.cels.2015.12.004">https://doi.org/10.1016/j.cels.2015.12.004</a>.
</div>
<div id="ref-Rodchenkov2019" class="csl-entry">
Rodchenkov, Igor, Ozgun Babur, Augustin Luna, Bulent Arman Aksoy,
Jeffrey V Wong, Dylan Fong, Max Franz, et al. 2019. <span>“<span
class="nocase">Pathway Commons 2019 Update: integration, analysis and
exploration of pathway data</span>.”</span> <em>Nucleic Acids
Research</em> 48 (D1): D489–97. <a
href="https://doi.org/10.1093/nar/gkz946">https://doi.org/10.1093/nar/gkz946</a>.
</div>
<div id="ref-Tercan2025" class="csl-entry">
Tercan, Bahar, Victor H Apolonio, Vinicius S Chagas, Christopher K Wong,
et al. 2025. <span>“Protocol for Assessing Distances in Pathway Space
for Classifier Feature Sets from Machine Learning Methods.”</span>
<em>STAR Protocols</em> 6 (2): 103681. https://doi.org/<a
href="https://doi.org/10.1016/j.xpro.2025.103681">https://doi.org/10.1016/j.xpro.2025.103681</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
