<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sysbiolab Team" />

<meta name="date" content="2025-10-23" />

<title>Visualizing spatial transcriptomics</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
<link href="site_libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script>
<script src="site_libs/bs3compat-0.9.0/transition.js"></script>
<script src="site_libs/bs3compat-0.9.0/tabs.js"></script>
<script src="site_libs/bs3compat-0.9.0/bs3compat.js"></script>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PathwaySpace</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="install.html">Install</a>
</li>
<li>
  <a href="get-started.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Introductory Vignettes</li>
    <li>
      <a href="projection-methods.html">PathwaySpace projection methods</a>
    </li>
    <li>
      <a href="modeling-signal-decay.html">Modeling signal decay functions</a>
    </li>
    <li>
      <a href="signal-aggregation-rules.html">Creating signal aggregation rules</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Visualizing features</li>
    <li>
      <a href="sparse-feature-sets.html">Sparse feature sets</a>
    </li>
    <li>
      <a href="spatial-transcriptomics.html">Spatial transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/sysbiolab/PathwaySpace">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Visualizing spatial transcriptomics</h1>
<h3 class="subtitle">Gene expression projections on spot
coordinates</h3>
<h4 class="author">Sysbiolab Team</h4>
<h4 class="date">2025-10-23</h4>

</div>


<p><strong>Package</strong>: PathwaySpace 1.0.3.1 <br/></p>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This vignette introduces the <em>SpotSpace</em> package, a
<em>PathwaySpace</em> extension for <em>Seurat</em> <span
class="citation">(Hao et al. 2024)</span>, providing methods for signal
propagation and visualization in spatial transcriptomics.
<em>SpotSpace</em> extends existing spatial analysis workflows to
explore signal patterns in tissue microenvironments. In what follows, we
present three step-by-step tutorials describing how to prepare input
data for <em>SpotSpace</em>. The results reproduce examples featured in
<em>Seurat</em>’s tutorials, so users are encouraged to see how these
packages can be used together.</p>
</div>
<div id="before-you-start" class="section level1">
<h1>Before You Start</h1>
<p>This vignette assumes prior experience with <a
href="https://satijalab.org/seurat/" target="_blank"
rel="noopener"><em>Seurat</em></a> <span class="citation">(Hao et al.
2024)</span>, especially for handling spatial transcriptomics data.</p>
<p><svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:orange;overflow:visible;position:relative;"><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"/></svg>
<strong>Note:</strong> If you are new to <em>Seurat</em>’s spatial
workflows, we recommend reviewing the <a
href="https://satijalab.org/seurat/articles/get_started_v5_new#spatial-analysis"
target="_blank" rel="noopener">spatial analysis tutorials</a> before
continuing.</p>
</div>
<div id="required-packages" class="section level1">
<h1>Required packages</h1>
<pre class="r"><code># Check required packages for this vignette
if (!require(&quot;remotes&quot;, quietly = TRUE)){
  install.packages(&quot;remotes&quot;)
}
if (!require(&quot;RGraphSpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (!require(&quot;PathwaySpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}
if (!require(&quot;SpotSpace&quot;, quietly = TRUE)){
  remotes::install_github(&quot;sysbiolab/SpotSpace&quot;)
}
if (!require(&quot;SeuratData&quot;, quietly = TRUE)){
  remotes::install_github(&quot;satijalab/seurat-data&quot;)
}</code></pre>
<pre class="r"><code># Check versions
if (packageVersion(&quot;RGraphSpace&quot;) &lt; &quot;1.1.0&quot;){
  message(&quot;Need to update &#39;RGraphSpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/RGraphSpace&quot;)
}
if (packageVersion(&quot;PathwaySpace&quot;) &lt; &quot;1.0.3.1&quot;){
  message(&quot;Need to update &#39;PathwaySpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/PathwaySpace&quot;)
}
if (packageVersion(&quot;SpotSpace&quot;) &lt; &quot;0.0.2&quot;){
  message(&quot;Need to update &#39;SpotSpace&#39; for this vignette&quot;)
  remotes::install_github(&quot;sysbiolab/SpotSpace&quot;)
}
if (packageVersion(&quot;Seurat&quot;) &lt; &quot;5.3.1.9999&quot;){
  message(&quot;Need to update &#39;Seurat&#39; for this vignette&quot;)
  remotes::install_github(&quot;satijalab/Seurat&quot;)
}</code></pre>
<pre class="r"><code># Load packages
library(RGraphSpace)
library(PathwaySpace)
library(SpotSpace)
library(Seurat)
library(SeuratObject)
library(SeuratData)
library(patchwork)</code></pre>
</div>
<div id="visium-v1-dataset" class="section level1">
<h1>Visium v1 dataset</h1>
<div id="setting-input-data" class="section level2">
<h2>Setting input data</h2>
<p>For this tutorial, we will use the <code>stxBrain</code> dataset from
the <em>SeuratData</em> package, consisting of spatial transcriptomics
data from sagittal mouse brain sections generated with Visium v1
technology. This dataset is commonly used to demonstrate <em>Seurat</em>
spatial workflows <span class="citation">(Hao et al. 2024)</span>. Here,
we will preprocess it with <em>Seurat</em> and then extract the relevant
data for <em>PathwaySpace</em> downstream analyses.</p>
<pre class="r"><code>## Install a Seurat dataset (this step is required only once)
SeuratData::InstallData(&quot;stxBrain&quot;)</code></pre>
<pre class="r"><code># Check manifest of installed datasets
# SeuratData::InstalledData()

# Load the &#39;stxBrain&#39; dataset
seurat_obj &lt;- LoadData(&quot;stxBrain&quot;, type = &quot;anterior1&quot;)</code></pre>
<p>The <code>stxBrain</code> dataset is normalized as suggested in
<em>Seurat</em>’s <a
href="https://satijalab.org/seurat/articles/spatial_vignette.html"
target="_blank" rel="noopener">spatial_vignette</a>, using the
<code>SCTransform()</code> function.</p>
<pre class="r"><code># Run vst normalization on counts
# seurat_obj &lt;- SCTransform(seurat_obj, assay = &quot;Spatial&quot;, verbose = FALSE)

# NOTE: Seurat recommends using SCTransform() for processing this 
# spatial dataset, which may require more computation time. Here,
# we use log-normalization for demonstration purposes.
seurat_obj &lt;- NormalizeData(seurat_obj)</code></pre>
<p>… and then we extract spot coordinates, tissue image, and
vst-normalized data.</p>
<pre class="r"><code># Get spot coordinates
spot_coord &lt;- GetTissueCoordinates(seurat_obj, scale = &quot;lowres&quot;)

# Get raster image
raster_image &lt;- GetImage(seurat_obj, &quot;raster&quot;)

# Get vst-normalized gene expression
vst_gexp &lt;- GetAssayData(seurat_obj, layer=&quot;data&quot;)

# If needed, remove seurat_obj to free memory
rm(seurat_obj)</code></pre>
<p>Next, we create a <em>PathwaySpace</em> object from the spot
coordinates and plot the resulting graph overlaid on the tissue
image.</p>
<pre class="r"><code># Create a PathwaySpace object from &#39;spot_coord&#39;, mapped to the &#39;raster_image&#39;
pspace_obj &lt;- buildSpotSpace(spot_coord = spot_coord, raster_image = raster_image)</code></pre>
<pre class="r"><code># Check spots on the top of the &#39;raster_image&#39;
xy_labs &lt;- labs(x=&quot;Spot coordinates 1&quot;, y=&quot;Spot coordinates 2&quot;)
plotGraphSpace(pspace_obj, add.image = TRUE) + xy_labs</code></pre>
<p><img src="figs_spot_space/fig1.png" width="100%" /></p>
</div>
<div id="running-pathwayspace" class="section level2">
<h2>Running <em>PathwaySpace</em></h2>
<p>Before projection, we need to specify a distance unit for the signal
decay function. This distance unit will affect the extent over which the
convolution operation projects the signal, scaled to the coordinate
space. We will use the center-to-center distance between spots, which
represents 100 µm in the Visium v1 technology.</p>
<pre class="r"><code># Get distance to the nearest spot
nspot &lt;- getNearestNode(pspace_obj)
pdist &lt;- mean(nspot$dist) # average distance
# &#39;pdist&#39; as the average center-to-center distance between spots
pdist
# [1] 0.01395144</code></pre>
<p>As an optional step, the <code>silhouetteMapping()</code> function
generates an image mask that outlines the graph layout, over which the
subsequent methods will project a landscape image. The
<code>baseline</code> argument controls the level at which a silhouette
is sliced to form the mask. Increasing the baseline (in
<code>[0,1]</code>) produces a more detailed, granular silhouette.</p>
<pre class="r"><code># Add a graph silhouette to the PathwaySpace object
pspace_obj &lt;- silhouetteMapping(pspace_obj, baseline = 0.1)
plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  add.image = TRUE, si.alpha = 0.5) + xy_labs</code></pre>
<p><img src="figs_spot_space/fig2.png" width="100%" /></p>
<p>Next, we specify the signal to be projected; for this demonstration,
we will use expression data from the <strong>Camk2n1</strong> gene. The
<code>vertexSignal()</code> accessor function is then used to assign the
gene expression values to graph vertices.</p>
<pre class="r"><code># Select a gene of interest (e.g., Camk2n1) and assign its 
# expression values to graph vertices
gene &lt;- &quot;Camk2n1&quot;
vertexSignal(pspace_obj)[colnames(vst_gexp)] &lt;- vst_gexp[gene,]</code></pre>
<p>We then perform the signal projection, setting
<code>decay = 0.5</code>. The decay parameter controls how the signal
attenuates as a function of distance in pathway space. With
<code>decay = 0.5</code>, the signal decreases to half of its initial
value at a distance equal to <code>pdist</code> (for additional
configuration details, see the <a
href="modeling-signal-decay.html"><em>modeling signal decay</em></a>
tutorial).</p>
<pre class="r"><code># Project gene signal
pspace_obj &lt;- circularProjection(pspace_obj, k = gs_vcount(pspace_obj), 
  decay.fun = weibullDecay(decay=0.5, pdist = pdist),
  aggregate.fun = signalAggregation(&quot;wmean&quot;))</code></pre>
<p>Because each spot produces an independent projection, the resulting
projections are aggregated into a unified landscape. Here we use a
weighted arithmetic mean, with each projection weighted by its own
magnitude (for additional configuration details, see the <a
href="signal-aggregation-rules.html"><em>signal aggregation
rules</em></a> tutorial).</p>
<p>Next, we show the results with minor variations to demonstrate some
of the available plot settings.</p>
<pre class="r"><code># Plot tissue image and projection separated 
p1 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, title = gene)
p1$image &lt;- p1$image + xy_labs
p1$graph &lt;- p1$graph + xy_labs

p1$image + p1$graph</code></pre>
<p><img src="figs_spot_space/fig3.png" width="100%" /></p>
<pre class="r"><code># Plot projections overlaid on the tissue image, with alpha = 0.25
p2 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, title = gene, 
  add.image = TRUE, si.alpha = 0.25) + xy_labs

# Plot projections overlaid on the tissue image, with zlim truncated at &gt;=0.5
p3 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, title = gene, 
  add.image = TRUE, si.alpha = 0.25, zlim = c(0.5, 1)) + xy_labs

p2 + p3</code></pre>
<p><img src="figs_spot_space/fig4.png" width="100%" /></p>
<p></br></p>
</div>
</div>
<div id="slide-seq-v2-dataset" class="section level1">
<h1>Slide-seq v2 dataset</h1>
<div id="setting-input-data-1" class="section level2">
<h2>Setting input data</h2>
<p>For this tutorial, we will use the <code>ssHippo</code> dataset
available from the <em>SeuratData</em> package, consisting of spatial
transcriptomics data from mouse hippocampus generated with
<strong>Slide-seq v2 technology</strong>. We will follow the same
general steps from our previous spatial tutorial, preprocessing with
<em>Seurat</em> and then extracting the relevant data for
<em>PathwaySpace</em> downstream analyses. For further details on this
dataset, see <em>Seurat</em>’s <a
href="https://satijalab.org/seurat/articles/spatial_vignette.html"
target="_blank" rel="noopener">spatial_vignette</a>.</p>
<pre class="r"><code>## Install a Seurat dataset (this step is required only once)
SeuratData::InstallData(&quot;ssHippo&quot;)</code></pre>
<pre class="r"><code># Check manifest of installed datasets
# SeuratData::InstalledData()

# Load the &#39;kidneyref&#39; dataset
seurat_obj &lt;- LoadData(&quot;ssHippo&quot;)</code></pre>
<pre class="r"><code># Run vst normalization on counts
# seurat_obj &lt;- SCTransform(seurat_obj, assay = &quot;Spatial&quot;, verbose = FALSE)

# NOTE: Seurat recommends using SCTransform() for processing this 
# spatial dataset, which may require more computation time. Here,
# we use log-normalization for demonstration purposes.
seurat_obj &lt;- NormalizeData(seurat_obj)</code></pre>
<pre class="r"><code># Extract spot coordinates and vst-normalized data
# Get spot coordinates
spot_coord &lt;- GetTissueCoordinates(seurat_obj)

#Note: the `ssHippo` dataset does not include a tissue image

# Get vst-normalized gene expression
vst_gexp &lt;- GetAssayData(seurat_obj, layer=&quot;data&quot;)

# If needed, remove seurat_obj to free memory
rm(seurat_obj)</code></pre>
<pre class="r"><code># Create a PathwaySpace object from &#39;spot_coord&#39;
pspace_obj &lt;- buildSpotSpace(spot_coord = spot_coord, nrc=700)</code></pre>
</div>
<div id="running-pathwayspace-1" class="section level2">
<h2>Running <em>PathwaySpace</em></h2>
<pre class="r"><code># Get distance to the nearest spot
nspot &lt;- getNearestNode(pspace_obj)
pdist &lt;- mean(nspot$dist) # average distance
# &#39;pdist&#39; as the average center-to-center distance between spots
pdist
# [1] 0.002497466</code></pre>
<pre class="r"><code># Add a graph silhouette to the PathwaySpace object
pspace_obj &lt;- silhouetteMapping(pspace_obj, fill.cavity = FALSE, 
  pdist = max(nspot$dist))

# Check silhouette plot
xy_labs &lt;- labs(x=&quot;Spot coordinates 1&quot;, y=&quot;Spot coordinates 2&quot;)
plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  add.image = TRUE, si.alpha = 0.5) + xy_labs</code></pre>
<p><img src="figs_spot_space/fig5.png" width="100%" /></p>
<pre class="r"><code># Choose a gene of interest (e.g., DDN) and assign its 
# expression values to graph vertices
gene &lt;- &quot;DDN&quot;
vertexSignal(pspace_obj)[colnames(vst_gexp)] &lt;- vst_gexp[gene,]

# Project gene signal
pspace_obj &lt;- circularProjection(pspace_obj, k = gs_vcount(pspace_obj), 
  decay.fun = weibullDecay(decay=0.5, pdist = pdist))

# Plot projections
#-- as a suggestion, truncate zlim at the upper limit 
#-- to enhance certain patters
p1 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  title = gene, zlim = c(0, 1)) + xy_labs</code></pre>
<pre class="r"><code># ...another gene (e.g. PCP4)
gene &lt;- &quot;PCP4&quot;
vertexSignal(pspace_obj)[colnames(vst_gexp)] &lt;- vst_gexp[gene,]

# Project gene signal
pspace_obj &lt;- circularProjection(pspace_obj, k = gs_vcount(pspace_obj), 
  decay.fun = weibullDecay(decay=0.5, pdist = pdist))

# Plot projections
p2 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  title = gene, zlim = c(0, 1)) + xy_labs</code></pre>
<pre class="r"><code>p1 + p2</code></pre>
<p><img src="figs_spot_space/fig6.png" width="100%" /></p>
</div>
</div>
<div id="visium-hd-dataset" class="section level1">
<h1>Visium HD dataset</h1>
<div id="setting-input-data-2" class="section level2">
<h2>Setting input data</h2>
<p>Here, we will use a higher-resolution spatial dataset from mouse
brain generated with <strong>Visium HD technology</strong>. This
platform provides whole-transcriptome gene expression data at a raw 2-µm
resolution, with additional binned versions available at 8 and 16 µm.
For this tutorial, we will use the 16-µm binned data. We will follow the
same general steps from our previous spatial tutorials, preprocessing
with <em>Seurat</em> and then extracting the relevant data for
<em>PathwaySpace</em> downstream analyses. For additional details on
this dataset, refer to <em>Seurat</em>’s <a
href="https://satijalab.org/seurat/articles/visiumhd_analysis_vignette.html"
target="_blank" rel="noopener">visiumhd_analysis_vignette</a>.</p>
<p>The Visium HD dataset can be downloaded from the 10x Genomics
repository at:</p>
<ul>
<li>Dataset name: Visium HD Spatial Gene Expression Library, Mouse Brain
(FFPE)</li>
<li>Repository URL: <a
href="https://www.10xgenomics.com/datasets/visium-hd-cytassist-gene-expression-libraries-of-mouse-brain-he"
target="_blank"
rel="noopener">https://www.10xgenomics.com/datasets</a></li>
<li>File name: Visium_HD_Mouse_Brain_binned_outputs.tar.gz</li>
<li>Size: 4.62 GB</li>
</ul>
<pre class="r"><code># Set &#39;localdir&#39; to the folder where the dataset was downloaded
# Use &#39;bin.size&#39; to choose the data resolution to load (2, 8, or 16 µm)
localdir &lt;- &quot;../Visium_HD_Mouse_Brain_binned_outputs&quot;
seurat_obj &lt;- Load10X_Spatial(data.dir = localdir, bin.size = 16)

# Check default assay
Assays(seurat_obj)
# [1] &quot;Spatial.016um&quot;</code></pre>
<pre class="r"><code># Run log-normalization for spatial data
seurat_obj &lt;- NormalizeData(seurat_obj)</code></pre>
<pre class="r"><code># Get spot coordinates
spot_coord &lt;- GetTissueCoordinates(seurat_obj, scale = &quot;lowres&quot;)

# Get raster image
raster_image &lt;- GetImage(seurat_obj, &quot;raster&quot;)

# Get normalized gene expression data
norm_gexp &lt;- GetAssayData(seurat_obj, layer=&quot;data&quot;)

# If needed, remove seurat_obj to free memory
rm(seurat_obj)</code></pre>
<pre class="r"><code># Create a PathwaySpace object from &#39;spot_coord&#39;, mapped to the &#39;raster_image&#39;
pspace_obj &lt;- buildSpotSpace(spot_coord = spot_coord, 
  raster_image = raster_image, nrc = 700)</code></pre>
</div>
<div id="running-pathwayspace-2" class="section level2">
<h2>Running <em>PathwaySpace</em></h2>
<pre class="r"><code># Get distance to the nearest spot
nspot &lt;- getNearestNode(pspace_obj)
pdist &lt;- mean(nspot$dist) # average distance
# &#39;pdist&#39; as the average center-to-center distance between spots
pdist
# [1] 0.002516144</code></pre>
<pre class="r"><code># Add a graph silhouette to the PathwaySpace object
pspace_obj &lt;- silhouetteMapping(pspace_obj, fill.cavity = FALSE, 
  pdist = max(nspot$dist))

# Check silhouette plot
xy_labs &lt;- labs(x=&quot;Spot coordinates 1&quot;, y=&quot;Spot coordinates 2&quot;)
plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  add.image = TRUE, si.alpha = 0.5) + xy_labs</code></pre>
<p><img src="figs_spot_space/fig7.png" width="100%" /></p>
<pre class="r"><code># Choose a gene of interest (e.g., Rorb) and assign its 
# expression values to graph vertices
gene &lt;- &quot;Rorb&quot;
vertexSignal(pspace_obj)[colnames(norm_gexp)] &lt;- norm_gexp[gene,]

# Project gene signal
pspace_obj &lt;- circularProjection(pspace_obj, k = gs_vcount(pspace_obj), 
  decay.fun = weibullDecay(decay=0.5, pdist = pdist))

# Plot projections
p1 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  title = gene, add.image = TRUE, zlim = c(0, 1)) + xy_labs</code></pre>
<pre class="r"><code># ...another gene (e.g. Hpca)
gene &lt;- &quot;Hpca&quot;
vertexSignal(pspace_obj)[colnames(norm_gexp)] &lt;- norm_gexp[gene,]

# Project gene signal
pspace_obj &lt;- circularProjection(pspace_obj, k = gs_vcount(pspace_obj), 
  decay.fun = weibullDecay(decay=0.5, pdist = pdist))

# Plot projections
p2 &lt;- plotPathwaySpace(ps=pspace_obj, theme = &quot;th3&quot;, 
  title = gene, add.image = TRUE, zlim = c(0, 1)) + xy_labs</code></pre>
<pre class="r"><code>p1 + p2</code></pre>
<p><img src="figs_spot_space/fig8.png" width="100%" /></p>
</div>
</div>
<div id="citation" class="section level1">
<h1>Citation</h1>
<p>If you use <em>PathwaySpace</em>, please cite:</p>
<ul>
<li><p>Tercan &amp; Apolonio et al. Protocol for assessing distances in
pathway space for classifier feature sets from machine learning methods.
<em>STAR Protocols</em> 6(2):103681, 2025. <a
href="https://doi.org/10.1016/j.xpro.2025.103681"
class="uri">https://doi.org/10.1016/j.xpro.2025.103681</a></p></li>
<li><p>Ellrott et al. Classification of non-TCGA cancer samples to TCGA
molecular subtypes using compact feature sets. <em>Cancer Cell</em>
43(2):195-212.e11, 2025. <a
href="https://doi.org/10.1016/j.ccell.2024.12.002"
class="uri">https://doi.org/10.1016/j.ccell.2024.12.002</a></p></li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<pre><code>## R version 4.5.1 (2025-06-13)
## Platform: x86_64-pc-linux-gnu
## Running under: Ubuntu 24.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: America/Sao_Paulo
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] Seurat_5.3.1.9999              SeuratObject_5.2.0            
##  [3] sp_2.2-0                       stxBrain.SeuratData_0.1.2     
##  [5] ssHippo.SeuratData_3.1.4       kidneyref.SeuratData_1.0.2    
##  [7] bonemarrowref.SeuratData_1.0.0 SeuratData_0.2.2.9002         
##  [9] fontawesome_0.5.3              patchwork_1.3.2               
## [11] igraph_2.2.0                   SpotSpace_0.0.2               
## [13] PathwaySpace_1.0.3.1           RGraphSpace_1.1.0             
## [15] ggplot2_4.0.0.9000             remotes_2.5.0                 
## [17] bs4cards_0.1.1                
## 
## loaded via a namespace (and not attached):
##   [1] deldir_2.0-4           pbapply_1.7-4          gridExtra_2.3         
##   [4] rlang_1.1.6            magrittr_2.0.4         RcppAnnoy_0.0.22      
##   [7] spatstat.geom_3.6-0    matrixStats_1.5.0      ggridges_0.5.7        
##  [10] compiler_4.5.1         png_0.1-8              vctrs_0.6.5           
##  [13] reshape2_1.4.4         stringr_1.5.2          crayon_1.5.3          
##  [16] pkgconfig_2.0.3        fastmap_1.2.0          promises_1.3.3        
##  [19] rmarkdown_2.30         purrr_1.1.0            xfun_0.53             
##  [22] cachem_1.1.0           jsonlite_2.0.0         goftest_1.2-3         
##  [25] later_1.4.4            spatstat.utils_3.2-0   irlba_2.3.5.1         
##  [28] parallel_4.5.1         cluster_2.1.8.1        R6_2.6.1              
##  [31] ica_1.0-3              spatstat.data_3.1-8    stringi_1.8.7         
##  [34] bslib_0.9.0            RColorBrewer_1.1-3     reticulate_1.43.0     
##  [37] spatstat.univar_3.1-4  parallelly_1.45.1      lmtest_0.9-40         
##  [40] jquerylib_0.1.4        scattermore_1.2        Rcpp_1.1.0            
##  [43] knitr_1.50             tensor_1.5.1           future.apply_1.20.0   
##  [46] zoo_1.8-14             sctransform_0.4.2      httpuv_1.6.16         
##  [49] Matrix_1.7-4           splines_4.5.1          tidyselect_1.2.1      
##  [52] abind_1.4-8            rstudioapi_0.17.1      yaml_2.3.10           
##  [55] spatstat.random_3.4-2  spatstat.explore_3.5-3 codetools_0.2-20      
##  [58] miniUI_0.1.2           listenv_0.9.1          lattice_0.22-5        
##  [61] tibble_3.3.0           plyr_1.8.9             shiny_1.11.1          
##  [64] withr_3.0.2            S7_0.2.0               ROCR_1.0-11           
##  [67] evaluate_1.0.5         Rtsne_0.17             future_1.67.0         
##  [70] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       
##  [73] fitdistrplus_1.2-4     pillar_1.11.1          KernSmooth_2.23-26    
##  [76] plotly_4.11.0          generics_0.1.4         RcppHNSW_0.6.0        
##  [79] scales_1.4.0           globals_0.18.0         xtable_1.8-4          
##  [82] glue_1.8.0             lazyeval_0.2.2         tools_4.5.1           
##  [85] data.table_1.17.8      RSpectra_0.16-2        RANN_2.6.2            
##  [88] fs_1.6.6               dotCall64_1.2          cowplot_1.2.0         
##  [91] grid_4.5.1             tidyr_1.3.1            nlme_3.1-168          
##  [94] cli_3.6.5              rappdirs_0.3.3         spatstat.sparse_3.1-0 
##  [97] spam_2.11-1            viridisLite_0.4.2      dplyr_1.1.4           
## [100] uwot_0.2.3             gtable_0.3.6           sass_0.4.10           
## [103] digest_0.6.37          progressr_0.17.0       ggrepel_0.9.6         
## [106] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     
## [109] lifecycle_1.0.4        httr_1.4.7             mime_0.13             
## [112] MASS_7.3-65</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-Yuhan2024" class="csl-entry">
Hao, Yuhan, Tim Stuart, Madeline H Kowalski, Saket Choudhary, Paul
Hoffman, Austin Hartman, Avi Srivastava, et al. 2024. <span>“Dictionary
Learning for Integrative, Multimodal and Scalable Single-Cell
Analysis.”</span> <em>Nature Biotechnology</em> 42 (2): 293–304. <a
href="https://doi.org/10.1038/s41587-023-01767-y">https://doi.org/10.1038/s41587-023-01767-y</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
